src/processors/default/processTrackArtworks.ts

```diff
@@ -0,0 +1,115 @@
+import { readFile, writeFile, rm } from 'node:fs/promises';
+import http from 'node:http';
+import https from 'node:https';
+import Path from 'node:path';
+
+import im from 'imagemagick';
+import { create } from 'ipfs-http-client';
+import tempDirectory from 'temp-dir';
+
+import { Table } from '../../db/db';
+import { missingProcessedArtworks } from '../../triggers/missing';
+import { Clients } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { rollPromises } from '../../utils/rollingPromises';
+
+const name = 'addMetadataIPFSHash';
+
+function imageBuffer(url: string): Promise<Buffer> {
+  console.log(`Downloading ${url}`)
+  return new Promise(function (resolve, reject) {
+    const client = url.includes('https') ? https : http;
+    client.get(url, function (res) {
+      const data: any = []
+      res.on('data', function (chunk) {
+        data.push(chunk);
+      }).on('end', function () {
+        const buffer = Buffer.concat(data);
+        resolve(buffer);
+      }).on('error', function (e) {
+        reject(e)
+      });
+    });
+  })
+}
+
+async function resizeImage(originalPath: string, size: string): Promise<Buffer> {
+  let resizedBuffer = null;
+  const buffer = await withTempPath(function (outPath) {
+    console.log(`resizing image to ${size} ${outPath}`)
+    return new Promise(function (resolve, reject) {
+      im.convert([originalPath, '-resize', size, outPath],
+        async function (err, stdout) {
+          if (err) reject(err);
+          resizedBuffer = await readFile(outPath)
+          resolve(resizedBuffer);
+        });
+    })
+  }, 'png');
+  return buffer;
+}
+
+async function uploadBuffer(buffer: Buffer) {
+  const url: any = '/ip4/127.0.0.1/tcp/5011'; // have to use any because create only accepts hash
```

Comment:
this needs to change to env var, Do we have a write url in env variables?

Reply:
sent to u directly

Reply:
thanks